airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -186,18 +186,18 @@ def ti_run(
         if not dr:
             raise ValueError(f"DagRun with dag_id={ti.dag_id} and run_id={ti.run_id} not found.")
 
-        # Clear XCom data for the task instance since we are certain it is executing
+        # Send the XCom data for clearance for the task instance since we are certain it is executing at this point.
```

Comment:
We should mention here why we can't do this on the server, namely we'll need to issue a purge in the custom XCom backend, and that the server may not have access or permission to do that.